### PR TITLE
feat: course generator apply new model

### PR DESCRIPTION
**Pull request by eugenio-d-tesio**
# 🤘 Courses PRs

## Description

Apply new model to generate json file output. This model group de courses and add the needed information to fetch it from the web site and generate all the courses structure without hardcoding pieces of it.

## Type

- [ ] Translation
- [x] Enhacement

### CODE REVIEW

File: .dart_tool/build_resolvers/sdk.sum.deps [hunk 1]
```diff
@@ -0,0 +1 @@
+{"sdk":"2.18.5 (stable) (Tue Nov 22 15:47:29 2022 +0000) on \"macos_arm64\"","analyzer":"/Users/eugeniotesio/.pub-cache/hosted/pub.dartlang.org/analyzer-5.6.0","build_resolvers":"/Users/eugeniotesio/.pub-cache/hosted/pub.dartlang.org/build_resolvers-2.2.0"}
```
Comment (EugenioTesio) [inline .dart_tool/build_resolvers/sdk.sum.deps:1]: ignore this

File: lib/json_generator.dart [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import 'dart:convert';
```
Comment (EugenioTesio) [inline lib/json_generator.dart:1]: Moved to lib folder and applied models

File: lib/json_generator.dart [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+import 'package:json_generator/models/_index.dart';
+
+void main() async {
+  var coursesDirectory = Directory('courses');
+  if (!await coursesDirectory.exists()) {
+    await coursesDirectory.create();
+  }
+
+  final List<FileCoursesGroup> coursesList = [
```
Comment (EugenioTesio) [inline lib/json_generator.dart:12]: `FileCoursesGroup` is the root model

File: courses/courses_es.json [hunk 4]
```diff
@@ -0,0 +1 @@
...,"description":"","contentUrl":"https://raw.githubusercontent.com/themonkslab/dream_teamers_courses/main/dart/10.Classes/5_Getters_y_setters.md","author":"The Monkslab","published":"2023-02-22T10:14:04.936263"},{"path":"10.Classes_6_Más_sobre_variables_y_methods","title":"6.Más sobre variables y methods","description":"","contentUrl":"https://raw.githubusercontent.com/themonkslab/dream_teamers_courses/main/dart/10.Classes/6_Más_sobre_variables_y_methods.md","author":"The Monkslab","published":"2023-02-22T10:14:04.936506"}]},{"path":"11.Medium_Dart","title":"11.Medium Dart","description":"","articles":[{"path":"11.Medium_Dart_1_Introduction","title":"1.Introduction","description":"","contentUrl":"https://raw.githubusercontent.com/themonkslab/dream_teamers_courses/main/dart/11.Medium_Dart/1_Introduction.md","author":"The Monkslab","published":"2023-02-22T10:14:04.924915"},{"path":"11.Medium_Dart_4_Debugging_a_project","title":"4.Debugging a project","description":"","contentUrl":"https://raw.githubusercontent.com/themonkslab/dream_teamers_courses/main/dart/11.Medium_Dart/4_Debugging_a_project.md","author":"The Monkslab","published":"2023-02-22T10:14:04.925087"}]}]}],"folder":"dart"}]
```
Comment (EugenioTesio) [inline courses/courses_es.json:1]: this is the result of the `json_generator.dart`. Use https://jsonformatter.org/ to prettify it.

File: pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  Just the pubspec for the json_generator.dart file
+version: 1.0.0
+
+environment:
+  sdk: '>=2.18.5 <3.0.0'
+  
+dev_dependencies:
+  build_runner: ^2.1.10
+  json_serializable: ^6.6.1
+  test: '>=1.15.0 <2.0.0'
```
Comment (EugenioTesio) [inline pubspec.yaml:12]: Add `pubspec.yaml` in order to allow import packages.

**Review by maurodibert — APPROVED** [review]

**Pull request merged**